corehq/apps/domain/decorators.py

```diff
@@ -84,6 +87,11 @@ def _inner(req, domain, *args, **kwargs):
                 else:
                     # some views might not have this set
                     couch_user = CouchUser.from_django_user(user)
+
+                dimagi_pages = [
+                    x for x in pages_not_restricted_for_dimagi if x % {'domain': domain_name} == req.path
+                ]
```

Comment:
I think you could combine this addition with the declaration above into a function with signature `_page_is_whitelisted(path, domain)` and then just drop that into the if statements which would probably be easier to read (and could be reused in both places).